### PR TITLE
Upgrade to python 3.10 and django 4

### PR DESCRIPTION
**Pull request by rebkwok**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,7 +6,7 @@ RUN npm ci
 
 RUN npm run build
 
-FROM python:3.9-buster
+FROM python:3.10-buster
```
Comment (rebkwok) [inline Dockerfile:9]: This doesn't use base-docker or the new docker way, but I plan to do that next :slightly_smiling_face:

Comment (ghickman) [reply Dockerfile:9]: @bloodearnest – can we target a particular version of python with the base-docker stack?

Comment (ghickman) [reply Dockerfile:9]: @bloodearnest – can we target a particular version of python with the base-docker stack?

**Review by ghickman — CHANGES_REQUESTED** [review]
Looks good.  You've missed some 3.9s in `.pre-commit-config.yml` though!

**Comment by rebkwok** [comment]
> Looks good. You've missed some 3.9s in `.pre-commit-config.yml` though!

ah. thanks, will get those now

**Review by ghickman — APPROVED** [review]

**Pull request merged**